### DISCUSSION TITLE
Access and use value from Query Row

### DISCUSSION THREAD

**Discussion by SaschaValentin** [Budibase/budibase#13218] [Help] [answered]
Under automations I have a query rows which returns only one row. This works all fine:

`{

    "rows": [
        {
            "vorname": "Rudi",
            "nachname": "Rüssel",
            "email": "rudi@web.de",
            "passwort": "ruessel",
            "rolle": "User",
            "_id": "%5B1%5D",
            "tableId": "datasource_plus_623776766_Person",
            "_rev": "rev" 
        } 
    ],
    "success": true 

}`

I can access the values either over {{ steps.1.rows.0.nachname }} in the bindings or with JS over
`var row = steps[1].rows[0].nachname;`

But how can I use these values in the design or for further working with it - for example write one to the localStorage. I´ve tried in JS scripting of automation but this doesn´t work.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @SaschaValentin,

So just for clarity, you want to run an automation and then use the output of that automation somewhere within your screen?

So in this case you likely are using some form of app trigger automation or some other variations. If you are using an automation like this. Then you would need to use a collect data step. This will allow you to do a last round of formatting, then whatever data this step returns can be used within your design screen once it gets the results back.

Theres a lot of information on this which can be found [here](https://docs.budibase.com/docs/synchronous-automations).
